data/transactions/logic/opcodeExplain.go

```diff
@@ -0,0 +1,277 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import "github.com/algorand/go-algorand/data/basics"
+
+// debugStackExplain explains the effect of an opcode over the stack
+// with 2 integers: deletions and additions, representing pops and inserts.
+// An opcode may delete a few variables from stack, then add a few to stack.
+type debugStackExplain func(*EvalContext) (int, int)
+
+// AppStateOperationT stands for the operation enum to app state, should be one of create, write, read, delete.
+type AppStateOperationT uint64
+
+const (
+	// AppStateWrite stands for writing to an app state.
+	AppStateWrite AppStateOperationT = iota + 1
+
+	// AppStateDelete stands for deleting an app state.
+	AppStateDelete
+)
+
+// AppStateEnum stands for the enum of app state type, should be one of global/local/box.
+type AppStateEnum uint64
+
+const (
+	// GlobalState stands for global state of an app.
+	GlobalState AppStateEnum = iota + 1
+
+	// LocalState stands for local state of an app.
+	LocalState
+
+	// Box stands for box storage of an app.
+	Box
+)
+
+// stateChangeExplain explains how an opcode change the app's state with a quadruple:
+// AppStateEnum stands for which app state: local/global/box,
+// AppStateOperationT stands for read/write/create/delete/check-existence,
+// together with key for touched state
+type stateChangeExplain func(ctx *EvalContext) (AppStateEnum, AppStateOperationT, basics.AppIndex, basics.TealValue, string)
```

Comment:
It seems a bit weird that the local state account is passed as a `basics.TealValue`

Reply:
for now I am returning the stack value representing the local state account directly, do you think it is ok to normalize them into 32-byte representation?